// @polkadot/apps-config/endpoints/index.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createWsEndpoints = createWsEndpoints;
Object.defineProperty(exports, "CUSTOM_ENDPOINT_KEY", {
  enumerable: true,
  get: function () {
    return _development.CUSTOM_ENDPOINT_KEY;
  }
});

var _development = require("./development.cjs");

var _production = require("./production.cjs");

var _testing = require("./testing.cjs");

var _testingRococo = require("./testingRococo.cjs");

// Copyright 2017-2021 @polkadot/apps-config authors & contributors
// SPDX-License-Identifier: Apache-2.0
function createWsEndpoints(t) {
  return [...(0, _development.createCustom)(t), {
    isDisabled: false,
    isHeader: true,
    text: t('rpc.header.live', 'Live networks', {
      ns: 'apps-config'
    }),
    textBy: '',
    value: ''
  }, ...(0, _production.createProduction)(t), {
    isDisabled: false,
    isHeader: true,
    text: t('rpc.header.test.relay', 'Test relays & parachains', {
      ns: 'apps-config'
    }),
    textBy: '',
    value: ''
  }, ...(0, _testingRococo.createRococo)(t), {
    isDisabled: false,
    isHeader: true,
    text: t('rpc.header.test', 'Test networks', {
      ns: 'apps-config'
    }),
    textBy: '',
    value: ''
  }, ...(0, _testing.createTesting)(t), {
    isDevelopment: true,
    isDisabled: false,
    isHeader: true,
    text: t('rpc.header.dev', 'Development', {
      ns: 'apps-config'
    }),
    textBy: '',
    value: ''
  }, ...(0, _development.createDev)(t), ...(0, _development.createOwn)(t)].filter(({
    isDisabled
  }) => !isDisabled);
}
