// @polkadot/apps-config/endpoints/index.js
// Copyright 2017-2021 @polkadot/apps-config authors & contributors
// SPDX-License-Identifier: Apache-2.0
import { createCustom, createDev, createOwn } from "./development.js";
import { createProduction } from "./production.js";
import { createTesting } from "./testing.js";
import { createRococo } from "./testingRococo.js";
export { CUSTOM_ENDPOINT_KEY } from "./development.js";
export function createWsEndpoints(t) {
  return [...createCustom(t), {
    isDisabled: false,
    isHeader: true,
    text: t('rpc.header.live', 'Live networks', {
      ns: 'apps-config'
    }),
    textBy: '',
    value: ''
  }, ...createProduction(t), {
    isDisabled: false,
    isHeader: true,
    text: t('rpc.header.test.relay', 'Test relays & parachains', {
      ns: 'apps-config'
    }),
    textBy: '',
    value: ''
  }, ...createRococo(t), {
    isDisabled: false,
    isHeader: true,
    text: t('rpc.header.test', 'Test networks', {
      ns: 'apps-config'
    }),
    textBy: '',
    value: ''
  }, ...createTesting(t), {
    isDevelopment: true,
    isDisabled: false,
    isHeader: true,
    text: t('rpc.header.dev', 'Development', {
      ns: 'apps-config'
    }),
    textBy: '',
    value: ''
  }, ...createDev(t), ...createOwn(t)].filter(({
    isDisabled
  }) => !isDisabled);
}
