// @polkadot/apps-config/api/spec/dotmog.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
// Copyright 2017-2021 @polkadot/apps-config authors & contributors
// SPDX-License-Identifier: Apache-2.0
// structs need to be in order

/* eslint-disable sort-keys */
const definitions = {
  types: [{
    // on all versions
    minmax: [0, undefined],
    types: {
      Address: 'MultiAddress',
      LookupSource: 'MultiAddress',
      MogwaiStruct: {
        id: 'Hash',
        dna: 'Hash',
        genesis: 'BlockNumber',
        price: 'Balance',
        gen: 'u64'
      },
      MogwaiBios: {
        mogwai_id: 'Hash',
        state: 'u32',
        metaxy: 'Vec<[u8;16]>',
        intrinsic: 'Balance',
        level: 'u8',
        phases: 'Vec<BlockNumber>',
        adaptations: 'Vec<Hash>'
      },
      GameEvent: {
        id: 'Hash',
        begin: 'BlockNumber',
        duration: 'u16',
        event_type: 'GameEventType',
        hashes: 'Vec<Hash>',
        value: 'u64'
      },
      GameEventType: {
        _enum: ['Default', 'Hatch']
      }
    }
  }]
};
var _default = definitions;
exports.default = _default;
