// @polkadot/apps-config/api/spec/index.js
// Copyright 2017-2021 @polkadot/apps-config authors & contributors
// SPDX-License-Identifier: Apache-2.0
import acala from "./acala.js";
import aresParachain from "./ares-parachain.js";
import bifrost from "./bifrost.js";
import bifrostParachain from "./bifrost-parachain.js";
import bitcountry from "./bitcountry.js";
import canvas from "./canvas.js";
import centrifugeChain from "./centrifuge-chain.js";
import chainx from "./chainx.js";
import clover from "./clover.js";
import cloverRococo from "./clover-rococo.js";
import crab from "./crab.js";
import crust from "./crust.js";
import crustParachain from "./crust-parachain.js";
import testPara from "./cumulus-test-parachain.js";
import darwinia from "./darwinia.js";
import darwiniaParachain from "./darwinia-parachain.js";
import datahighwayParachain from "./datahighway.js";
import dock from "./dock.js";
import dotmog from "./dotmog.js";
import dusty from "./dusty.js";
import edgeware from "./edgeware.js";
import encointerNodeNotee from "./encointer-node-notee.js";
import encointerNodeTeeproxy from "./encointer-node-teeproxy.js";
import encointerPara from "./encointer-para.js";
import equilibrium from "./equilibrium.js";
import hanonycash from "./hanonycash.js";
import hydrate from "./hydrate.js";
import idavoll from "./idavoll.js";
import integritee from "./integritee.js";
import jupiter from "./jupiter.js";
import jupiterRococo from "./jupiter-rococo.js";
import kilt from "./kilt.js";
import kulupu from "./kulupu.js";
import laminar from "./laminar.js";
import litentry from "./litentry.js";
import moonbeam from "./moonbeam.js";
import nodle from "./nodle.js";
import phala from "./phala.js";
import phalaParachain from "./phala-parachain.js";
import plasm from "./plasm.js";
import plasmParachain from "./plasm-parachain.js";
import polkabtc from "./polkabtc.js";
import polkadex from "./polkadex.js";
import robonomics from "./robonomics.js";
import sgc from "./sgc.js";
import snowbridge from "./snowbridge.js";
import soraSubstrate from "./soraSubstrate.js";
import stafi from "./stafi.js";
import subdao from "./subdao.js";
import subsocial from "./subsocial.js";
import ternoa from "./ternoa.js";
import trustbase from "./trustbase.js";
import uniarts from "./uniarts.js";
import unique from "./unique.js";
import zenlink from "./zenlink.js";
import zero from "./zero.js"; // NOTE: The mapping is done from specName in state.getRuntimeVersion

const spec = {
  Crab: crab,
  Darwinia: darwinia,
  Equilibrium: equilibrium,
  acala,
  'ares-parachain': aresParachain,
  bifrost: bifrost,
  'bifrost-parachain': bifrostParachain,
  'bitcountry-node': bitcountry,
  'btc-parachain': polkabtc,
  canvas,
  'centrifuge-chain': centrifugeChain,
  chainx,
  clover,
  'clover-rococo': cloverRococo,
  crust,
  'crust-parachain': crustParachain,
  'cumulus-test-parachain': testPara,
  'darwinia-parachain': darwiniaParachain,
  'datahighway-parachain': datahighwayParachain,
  'dock-main-runtime': dock,
  'dock-testnet': dock,
  'dotmog-node': dotmog,
  dusty3: dusty,
  edgeware,
  'encointer-node-notee': encointerNodeNotee,
  'encointer-node-teeproxy': encointerNodeTeeproxy,
  'encointer-parachain': encointerPara,
  'hack-hydra-dx': hydrate,
  hanonycash,
  'hydra-dx': hydrate,
  idavoll,
  'integritee-parachain': integritee,
  'jupiter-prep': jupiter,
  'jupiter-rococo': jupiterRococo,
  'kilt-parachain': kilt,
  kulupu,
  laminar,
  litentry,
  mandala: acala,
  'mashnet-node': kilt,
  'moonbase-alphanet': moonbeam,
  moonbeam,
  'moonbeam-standalone': moonbeam,
  nft: unique,
  'node-moonbeam': moonbeam,
  'node-polkadex': polkadex,
  'nodle-chain': nodle,
  'phala-collator': phalaParachain,
  'phala-node': phala,
  plasm,
  'plasm-parachain': plasmParachain,
  robonomics,
  'sgc-node': sgc,
  snowbridge,
  'sora-substrate': soraSubstrate,
  stafi,
  subdao,
  subsocial,
  subzero: zero,
  ternoa,
  trustbase,
  uniarts,
  zenlink
};
export default spec;
